cucumber-expressions/javascript/test/cucumber_expression_test.js

```diff
@@ -63,7 +63,7 @@ describe(CucumberExpression.name, () => {
   })
 
   describe('RegExp special characters', () => {
-    ['[', '\\', '^', '$', '?', '*', '+'].forEach((character) => {
+    ['\\', '[', ']', '^', '$', '.', '|', '?', '*', '+'].forEach((character) => {
```

Comment:
I don't think this change is needed. Moving the backslash to first is fine, but the addition of the closing square bracket, dot, and pipe are unneeded. 

The dot and pipe are tested separately as these tests weren't failing before implement escaping. 

The closing square bracket doesn't need to be tested as that too probably isn't failing and looking up regular expression special characters, it didn't appear that you need to escape the closing square bracket.